spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -431,9 +436,12 @@ else if (theListener instanceof BatchAcknowledgingMessageListener) {
 			else {
 				throw new IllegalArgumentException("Listener must be one of 'MessageListener', "
 						+ "'BatchMessageListener', 'AcknowledgingMessageListener', "
-						+ "'BatchAcknowledgingMessageListener', not " + theListener.getClass().getName());
+						+ "'BatchAcknowledgingMessageListener', not " + this.theListener.getClass().getName());
 			}
 			Assert.state(!this.isBatchListener || !this.isRecordAck, "Cannot use AckMode.RECORD with a batch listener");
+			if (this.autoCommit && this.theListener instanceof ConsumerSeekAware) {
+				((ConsumerSeekAware) this.theListener).registerSeekCallback(this);
```

Comment:
1. I wasn't able to step in here in debug mode during testing. Maybe we just don't have any test on the matter?..
2. Having your condition, "the listener is invoked on the consumer thread."  and "listeners should store the callback in a `{@code ThreadLocal}`.", plus the test-case in the `EnableKafkaIntegrationTests` with exactly a `ThreadLocal`, I dug further and found this code in the `KafkaMessageListenerContainer.doStart()`:
```
this.listenerConsumer = new ListenerConsumer(this.listener, this.acknowledgingMessageListener);
		setRunning(true);
		this.listenerConsumerFuture = containerProperties
				.getConsumerTaskExecutor()
				.submitListenable(this.listenerConsumer);
```
So, we create `ListenerConsumer` instance in the `KafkaMessageListenerContainer.doStart()` Thread, but `run()` it in a different Thread.

My concern that this `.registerSeekCallback(this)` for the `autoCommit = true` is done in a wrong Thread. Therefore I won't be able to get a proper `ThreadLocal` value, when I consume messages.

Am I missing anything?
Thanks